driver-sync/src/test/functional/com/mongodb/client/AtlasDataLakeTest.java

```diff
@@ -0,0 +1,337 @@
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+
+// See https://github.com/mongodb/specifications/tree/master/source/transactions/tests
+@RunWith(Parameterized.class)
+public class AtlasDataLakeTest {
+    private final String filename;
+    private final String description;
+    private final String databaseName;
+    private final BsonArray data;
+    private final BsonDocument definition;
+    private final boolean skipTest;
+    private JsonPoweredCrudTestHelper helper;
+    private final TestCommandListener commandListener;
+    private final TestConnectionPoolListener connectionPoolListener;
+    private MongoClient mongoClient;
+    private CollectionHelper<Document> collectionHelper;
+    private ConnectionString connectionString = null;
+    private final String collectionName;
+    private MongoDatabase database;
+
+    public AtlasDataLakeTest(final String filename, final String description, final String databaseName, final String collectionName,
+                             final BsonArray data, final BsonDocument definition, final boolean skipTest) {
+        this.filename = filename;
+        this.description = description;
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.data = data;
+        this.definition = definition;
+        this.commandListener = new TestCommandListener();
+        this.connectionPoolListener = new TestConnectionPoolListener();
+        this.skipTest = skipTest;
+    }
+
+    @Before
+    public void setUp() {
+        assumeFalse(skipTest);
+
+        collectionHelper = new CollectionHelper<Document>(new DocumentCodec(), new MongoNamespace(databaseName, collectionName));
+        connectionString = getADLConnectionString();
+
+        MongoClientSettings.Builder builder = getMongoClientSettingsBuilder()
+                .applyConnectionString(connectionString)
+                .addCommandListener(commandListener)
+                .retryWrites(false)
+                .retryReads(false);
+        mongoClient = createMongoClient(builder.build());
+
+        database = mongoClient.getDatabase(databaseName);
+        helper = new JsonPoweredCrudTestHelper(description, database, database.getCollection(collectionName, BsonDocument.class),
+                null, mongoClient);
+    }
+
+    @After
+    public void cleanUp() {
+        if (mongoClient != null) {
+            mongoClient.close();
+        }
+    }
+
+    @Test
+    public void shouldPassAllOutcomes() {
+        if (definition.containsKey("uri")) {
+            if (definition.getBoolean("valid").getValue()) {
+                testValidUri(definition.getString("uri").getValue());
+            } else {
+                testInvalidUri(definition.getString("uri").getValue());
+            }
+        } else {
+            executeOperations(definition.getArray("operations"));
+
+            if (definition.containsKey("expectations")) {
```

Comment:
Do any of the JSON files have an `expectations` key?  If not, this can be removed.